.storybook/blocks/ComponentDetails.jsx

```diff
@@ -119,6 +157,175 @@ const VersionDetails = ({ tag, data = {}, isDeprecated = false, skipDate = false
+			</svg>
+		</ResourceIconWrapper>
+	)
+};
+
+const adobeSvg = () => {
+	return (
+		<ResourceIconWrapper>
+			<svg viewBox="0 0 36 36" focusable="false" aria-hidden="true" aria-label="AdobeLogo">
+				<path fill="rgb(250, 15, 0)" d="M22.175 4H34v28L22.175 4zm-8.336 0H2v28L13.839 4zm4.165 10.317l7.538 17.682h-4.939l-2.258-5.632h-5.517l5.176-12.05z"></path>
+			</svg>
+		</ResourceIconWrapper>
+	)
+};
+
+/**
+ * Determines which SVG to use based on the resource link
+ */
+const iconSvgs = (linkType) => {
+	let svgAsset;
+
+	switch(linkType) {
+		case "package": 
+			svgAsset = npmSvg()
+			break;
+		case "repository": 
+			svgAsset = githubSvg()
+			break;
+		case "guidelines": 
+			svgAsset = adobeSvg()
+			break;
+		default: 
+			break;
+	}
+
+	return svgAsset;
+};
+
+/**
+ * Converts the linkType to the subtitle text
+ */
+const convertLinkTypeText = (linkType) => {
+	switch(linkType) {
+		case "package": 
+			linkType = "npm"
+			break;
+		case "repository": 
+			linkType = "GitHub"
+			break;
+		case "guidelines": 
+			linkType = "Spectrum website"
+			break;
+		default: 
+			break;
+	}
+	return linkType;
+};
+
+export const ResourceLinkContent = ({data, linkType=["package", "repository", "guidelines"]}) => {
+	const packageJson = data;
+
+	// componentGuidelinesName
+	let packageName = "";
+	let packageLink = "";
+
+	// componentBetaName: for components with guidelines that are still on spectrum-contributions/beta site
+	let packageAltName = "";
+	let packageAltLink = "";
+	
+	if(linkType === "package") {
+		// NPM package name and link 
+		packageName = packageJson?.name ?? undefined;
+		packageLink = (packageName && typeof packageName !== "undefined") ? `https://npmjs.com/${packageName}` : false;
+	}
+	
+	else if (linkType === "repository") {
+		// repo name and link 
+		packageName = packageJson?.name ? packageJson?.name.split('/').pop() : undefined;
+		packageLink = (packageName && typeof packageName !== "undefined") ? `https://github.com/adobe/spectrum-css/tree/main/components/${packageName}` : false;
+	}
+
+	else if (linkType === "guidelines") {
+		// guidelines site name and link 
+		packageName = packageJson?.componentGuidelinesName ?? undefined;
+		packageLink = (packageName && typeof packageName !== "undefined") ? `https://spectrum.adobe.com/page/${packageName}` : false;
+
+		// internal contributions/beta guidelines name and link
+		packageAltName = (packageName) ? undefined : packageJson?.componentBetaName;
+
+		if (!packageLink) {
+			packageAltLink = (packageAltName && typeof packageAltName !== "undefined") ? `https://spectrum-contributions.corp.adobe.com/page/${packageAltName}` : false;
+			packageLink = packageAltLink;
+		}
+	}
```

Comment:
It definitely feels like maybe I could refactor and get rid of some of the repetition. Any suggestions as to how I could do that? Maybe this is ok?